parsing/parser.mly

```diff
@@ -336,24 +334,27 @@ let lapply ~loc p1 p2 =
   else raise (Syntaxerr.Error(
                   Syntaxerr.Applicative_path (make_loc loc)))
 
-let exp_of_longident ~loc lid =
-  mkexp ~loc (Pexp_ident {lid with txt = Lident(Longident.last lid.txt)})
-
 (* [loc_map] could be [Location.map]. *)
-let loc_map (f : 'a -> 'b) (x : 'a Location.loc) : 'b Location.loc =
-  { x with txt = f x.txt }
+let loc_map ?(mk_ghost=false) (f : 'a -> 'b) (x : 'a Location.loc)
+  : 'b Location.loc =
+  { txt = f x.txt
+  ; loc = if not mk_ghost then x.loc else { x.loc with loc_ghost = true }}
```

Comment:
I'm not fond of this API. You are passing extra information within the function call for something that you could just as easily change from the outside: instead of `loc_map ~mk_ghost:true f x`, just write `make_ghost (loc_map f x)`.